src/source-af-packet.c

```diff
@@ -1388,14 +1548,168 @@ frame size: TPACKET_ALIGN(snaplen + TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + siz
     return 1;
 }
 
+static int AFPComputeRingParamsV3(AFPThreadVars *ptv)
+{
+    ptv->req3.tp_block_size = getpagesize();
+    ptv->req3.tp_frame_size = 2048;
+    int frames_per_block = ptv->req3.tp_block_size / ptv->req3.tp_frame_size;
+    if (frames_per_block == 0) {
+        SCLogInfo("frame size to big");
+        return -1;
+    }
+    ptv->req3.tp_frame_nr = ptv->ring_size;
+    ptv->req3.tp_block_nr = ptv->req3.tp_frame_nr / frames_per_block + 1;
+    /* exact division */
+    ptv->req3.tp_frame_nr = ptv->req3.tp_block_nr * frames_per_block;
+    ptv->req3.tp_retire_blk_tov = 10; /* 10 ms timeout on block */
```

Comment:
In TPACKET_V3, blocks are rotated always after the timeout expires. To be able to really tune for performance, this timeout must be configurable. The mechanism is fundamentally different from TPACKET_V2. See previous comment.